pkg/api/errors/state.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package errors
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"google.golang.org/grpc/codes"
+
+	kiterrors "github.com/dapr/kit/errors"
+)
+
+func StateStoreNotConfigured() error {
+	return kiterrors.NewBuilder(
+		codes.FailedPrecondition,
+		http.StatusInternalServerError,
+		"state store is not configured",
+		"ERR_STATE_STORE_NOT_CONFIGURED",
+	).
+		WithErrorInfo(kiterrors.CodePrefixStateStore+kiterrors.CodeNotConfigured, nil).
+		Build()
+}
+
+func StateStoreNotFound(storeName string) error {
+	return kiterrors.NewBuilder(
+		codes.InvalidArgument, // TODO We should change this at some point. It should be codes.NotFound, but it will be a breaking change.
+		http.StatusBadRequest,
+		fmt.Sprintf("state store %s is not found", storeName),
+		"ERR_STATE_STORE_NOT_FOUND",
+	).
+		WithErrorInfo(kiterrors.CodePrefixStateStore+kiterrors.CodeNotFound, nil).
+		Build()
+}
+
+func StateStoreInvalidKeyName(storeName string, key string, msg string) error {
+	return kiterrors.NewBuilder(
+		codes.InvalidArgument,
+		http.StatusBadRequest,
+		msg,
+		"ERR_MALFORMED_REQUEST",
+	).WithErrorInfo(kiterrors.CodePrefixStateStore+kiterrors.CodeIllegalKey, nil).
+		WithResourceInfo("state", storeName, "", "").
+		WithFieldViolation(key, msg).
+		Build()
+}
+
+/**** Transactions ****/
+
+func StateStoreTransactionsNotSupported(storeName string) error {
+	return kiterrors.NewBuilder(
+		codes.Unimplemented,
+		http.StatusInternalServerError,
+		fmt.Sprintf(MsgStateTransactionsNotSupported, storeName),
```

Comment:
Actually, I think this const should just live here in this func since it's only used once as far as my ctrl-f can find